server/repository/src/db_diesel/program_row.rs

```diff
@@ -0,0 +1,72 @@
+use super::StorageConnection;
+
+use crate::repository_error::RepositoryError;
+
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+
+table! {
+    program (id) {
+        id -> Text,
+        #[sql_name = "type"] type_ -> Text,
+        name -> Text,
+        patient_id -> Text,
+        enrolment_datetime -> Timestamp,
+        program_patient_id -> Nullable<Text>,
+    }
+}
+
+#[derive(Clone, Insertable, Queryable, Debug, PartialEq, Eq, AsChangeset)]
+#[table_name = "program"]
+pub struct ProgramRow {
+    /// The row id
+    pub id: String,
+    /// The type of the program, same as the matching program document type.
+    #[column_name = "type_"]
+    pub r#type: String,
+    /// The program document name
+    pub name: String,
+    /// The patient this program belong to
```

Comment:
:nit - this is very nit-picky, sorry.
```suggestion
    /// The patient this program belongs to
```